src/JustSaying/Fluent/QueueAddressQueue.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using Amazon;
+using Amazon.SQS;
+using JustSaying.AwsTools.MessageHandling;
+using JustSaying.Messaging.Interrogation;
+
+namespace JustSaying.Fluent
+{
+    /// <summary>
+    /// A queue that is constructed from a Queue URL, that has no behaviour.
+    /// </summary>
+    internal sealed class QueueAddressQueue : ISqsQueue
+    {
+        public QueueAddressQueue(QueueAddress queueAddress, IAmazonSQS sqsClient)
+        {
+            Uri = queueAddress.QueueUrl;
+            var pathParts = queueAddress.QueueUrl.AbsolutePath.Split(new[] { '/' }, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
Could you use `Uri.Segments` to do this rather than split the string, or does that return too many bits?

Reply:
I forgot all about `Uri.Segments`, will update